ServantHDBC/src/Api/Authentication.hs

```diff
@@ -10,22 +10,36 @@
 module Api.Authentication (AuthAPI, authHandlers) where
 
 import Control.Monad.Trans.Except
+import Data.Aeson
+import GHC.Generics
 import Servant
 
-type Auth = String
-type AuthAPI = "sessions" :>
-                         ("new" :> Post '[JSON] Auth
-                          :<|> "refresh" :> Post '[JSON] Auth
-                          :<|> "destroy" :> Post '[JSON] Auth)
+data Auth = Auth String
+    deriving (Generic)
 
-authNew :: ExceptT ServantErr IO Auth
-authNew = return "should create a cookie"
+instance FromJSON Auth
 
-authRefresh :: ExceptT ServantErr IO Auth
-authRefresh = return "should refresh session timeout"
+type AuthAPI = "sessions" :> AuthAPI'
 
-authDestroy :: ExceptT ServantErr IO Auth
-authDestroy = return "should destroy the session"
+type AuthAPI' = 
+            "new" 
+                :> ReqBody '[JSON] Auth
+                    :> Post '[JSON] String
+       :<|> "refresh" 
+                :> ReqBody '[JSON] Auth
+                    :> Post '[JSON] String
+       :<|> "destroy" 
+                :> ReqBody '[JSON] Auth
+                    :> Post '[JSON] String
```

Comment:
I don't know. The tutorial tends to put one handler on one line (http://haskell-servant.readthedocs.io/en/stable/tutorial/Server.html) That's ok for relatively type components,  say 5. When i got to the longer methods like productFilter, having everything on one long line is hard to deal with.

Doing it this way, the path component has the least amount of indenting, it makes a table of contents kind of structure. then next level of indention are the captured variables and finally the deepest indention is the result. I can tell at a glance what i'm expecting for url components.

Finally, i put the separators first to allow easy commenting in and out of various components, much like record syntax.

I usually try to match my formatting to whatever the rest of the code base looks like. In this case, i felt i was free to do whatever worked for me. Perhaps it's not a good style, but it worked well for me, and is internally consistent across all of the files.

If there's another format you prefer, point me at an example (a long example would be nice, something like filterProducts) and i'll go back and reformat to match that style.